spec/models/storage_spec.rb

```diff
@@ -505,4 +505,24 @@
       expect(storage.smartstate_analysis_count_for_host_id(host.id)).to eq(2)
     end
   end
+
+  context '#storage_clusters' do
+    it 'returns only parents of type storage_cluster' do
+      # A storage mounted on different VCs will have multiple parents
+      storage          = FactoryGirl.create(:storage,               :name => 'test_storage_vmfs', :store_type => 'VMFS')
+      ems_folder       = FactoryGirl.create(:ems_folder,            :name => 'test_folder')
+      storage_cluster1 = FactoryGirl.create(:storage_cluster,       :name => 'test_storage_cluster1')
+      storage_cluster2 = FactoryGirl.create(:storage_cluster,       :name => 'test_storage_cluster2')
+      ems_folder.add_child(storage)
+      storage_cluster1.add_child(storage)
+      storage_cluster2.add_child(storage)
+      expect(storage.storage_clusters.to_set).to eq([storage_cluster1, storage_cluster2].to_set)
+      expect(storage.parents.to_set).to eq([ems_folder, storage_cluster1, storage_cluster2].to_set)
```

Comment:
no need to use .to_set here. check `contain_exactly` or `match_array` matcher [here](https://www.relishapp.com/rspec/rspec-expectations/v/3-4/docs/built-in-matchers/contain-exactly-matcher)